src/app/pages/examiner-stats/examiner-stats.analytics.effects.ts

```diff
@@ -0,0 +1,130 @@
+import { of } from 'rxjs';
+import { AnalyticRecorded } from '@providers/analytics/analytics.actions';
+import {
+  AccordionClosed,
+  AccordionOpened,
+  ColourFilterChanged,
+  DateRangeChanged,
+  ExaminerStatsViewDidEnter,
+  HideChartsActivated,
+  HideChartsDeactivated,
+  LocationChanged,
+  TestCategoryChanged,
+} from '@pages/examiner-stats/examiner-stats.actions';
+
+@Injectable()
+export class ExaminerStatsAnalyticsEffects {
+
+  constructor(
+    private analytics: AnalyticsProvider,
+    private actions$: Actions,
+  ) {
+  }
+
+  examinerStatsViewDidEnter$ = createEffect(() => this.actions$.pipe(
+    ofType(ExaminerStatsViewDidEnter),
+    switchMap(() => {
+      this.analytics.setCurrentPage(AnalyticsScreenNames.EXAMINER_STATS);
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  dateRangeChanged$ = createEffect(() => this.actions$.pipe(
+    ofType(DateRangeChanged),
+    switchMap(({ range }) => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.DATE_RANGE_CHANGED,
+        range,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  locationChanged$ = createEffect(() => this.actions$.pipe(
+    ofType(LocationChanged),
+    switchMap(({ location }) => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.LOCATION_CHANGED,
+        location,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  testCategoryChanged$ = createEffect(() => this.actions$.pipe(
+    ofType(TestCategoryChanged),
+    switchMap(({ testCategory }) => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.TEST_CATEGORY_CHANGED,
+        testCategory,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  colourFilterChanged$ = createEffect(() => this.actions$.pipe(
+    ofType(ColourFilterChanged),
+    switchMap(({ colour }) => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.COLOUR_SCHEME_CHANGED,
+        colour,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  accordionOpened$ = createEffect(() => this.actions$.pipe(
+    ofType(AccordionOpened),
+    switchMap(() => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.ADDITIONAL_FILTERS_TOGGLED,
+        'Additional filters opened',
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+  accordionClosed$ = createEffect(() => this.actions$.pipe(
+    ofType(AccordionClosed),
+    switchMap(() => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.EXAMINER_STATS,
+        AnalyticsEvents.ADDITIONAL_FILTERS_TOGGLED,
+        'Additional filters closed',
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
```

Comment:
This can once again be simplified, instead of creating an action for open and close, create one that takes in the status and then based upon that vary the `opened` / `closed` message